internal/cmd/kafka/command_cluster_create.go

```diff
@@ -345,7 +288,7 @@ func setCmkClusterConfig(typeString string, cku int32, encryptionKeyID string) *
 		}
 	case skuDedicated:
 		return &cmkv2.CmkV2ClusterSpecConfigOneOf{
-			CmkV2Dedicated: &cmkv2.CmkV2Dedicated{Kind: "Dedicated", Cku: cku, EncryptionKey: &encryptionKeyID},
+			CmkV2Dedicated: &cmkv2.CmkV2Dedicated{Kind: "Dedicated", Cku: cku, EncryptionKey: encryptionKeyID},
```

Comment:
The code is a bit cleaner if we wait to convert this to a pointer, IMO.

Reply:
Moved the conversion to ptr into the `setCmkClusterConfig` function.